pybamm/simulation.py

```diff
@@ -775,6 +810,52 @@ def solve(
                             # Otherwise, just stop this cycle
                             break
 
+                    step_termination = step_solution.termination
+
+                    # Add a padding rest step if necessary
+                    if op_conds["next timestamp"] is not None:
+                        rest_time = (
+                            op_conds["next timestamp"]
+                            - (
+                                self.experiment.initial_timestamp
+                                + timedelta(seconds=float(step_solution.t[-1]))
+                            )
+                        ).total_seconds()
+                        if rest_time > 0:
+                            start_time = step_solution.t[-1]
+                            # Let me know if you have a better name
+                            op_conds_str = "Rest for padding"
+                            model = self.op_conds_to_built_models[op_conds_str]
+                            solver = self.op_conds_to_built_solvers[op_conds_str]
+
+                            logs["step number"] = (step_num, cycle_length)
+                            logs["step operating conditions"] = op_conds_str
+                            callbacks.on_step_start(logs)
+
+                            ambient_temp = (
+                                op_conds["temperature"] or self._original_temperature
+                            )
+                            kwargs["inputs"] = {
+                                **user_inputs,
+                                **op_conds,
+                                "Ambient temperature [K]": ambient_temp,
+                                "start time": start_time,
+                            }
+                            # Make sure we take at least 2 timesteps
+                            # The period is hardcoded to 10 minutes, the user can
+                            # always override it by adding a rest step
+                            npts = max(int(round(rest_time / 600)) + 1, 2)
```

Comment:
Not sure what's the better approach here. Another option would be to take 10 (or 20) steps or 10 min steps, whatever results in fewer steps.